src/abstract_calc.h

```diff
@@ -0,0 +1,16 @@
+#ifndef ABSTRACT_CALC_H
+#define ABSTRACT_CALC_H
+
+#include "ast.h"
+#include "config.h"
+#include "lexer.h"
+
+typedef struct abstract_calc_t {
+    char *expr;
+    calc_value_t result;
+
+    int (*run) (struct abstract_calc_t *calc);
```

Comment:
struct is optional here just because you have a typedef that treats struct abstract_calc_t as abstract_calc_t